File: 1-Layers/1.1-Presentation/RoleWebApi.Client.Web/ClientApp/src/app/models/adapters/client-adapter.ts
import { Injectable } from "@angular/core";
import { ClientDTO } from "../dtos/index";
import { ClientListCollectionViewModel } from "../../modules/policy/viewmodels";

@Injectable()
export class ClientAdapter {
  constructor() {
  }

  ClientToListCollectionViewModel(client: ClientDTO): ClientListCollectionViewModel {

    let clientListCollectionVM: ClientListCollectionViewModel = new ClientListCollectionViewModel();

    clientListCollectionVM.Id = client.Id;
    clientListCollectionVM.Name = client.Name;
    clientListCollectionVM.Email = client.Email;
    clientListCollectionVM.Role = client.Role;

    return clientListCollectionVM;
  }

  ClientsToListCollectionsViewModel(clients: ClientDTO[]): ClientListCollectionViewModel[] {

    let _self = this;
    let clientListCollectionsVM: ClientListCollectionViewModel[] = new Array<ClientListCollectionViewModel>();

    clients.forEach(function (client) {
      clientListCollectionsVM.push(_self.ClientToListCollectionViewModel(client));
    });

    return clientListCollectionsVM;
  }
}

File: 1-Layers/1.3-Application/RoleWebApi.Application.Core/Adapters/ClientAdapter.cs
namespace RoleWebApi.Application.Core.Adapters
{
    using RoleWebApi.Infrastructure.CrossCutting.DTO;
    using RoleWebApi.Infrastructure.Data.Entity.Domain;
    using System.Collections.Generic;

    public class ClientAdapter : IClientAdapter
    {
        public IEnumerable<ClientDTO> FromClientsToClientsDTO(ICollection<Client> clients)
        {
            if (clients == null)
                return null;

            List<ClientDTO> clientsDTO = new List<ClientDTO>();

            foreach (Client client in clients)
            {
                clientsDTO.Add(FromClientToClientDTO(client));
            }

            return clientsDTO;
        }

        public ClientDTO FromClientToClientDTO(Client client)
        {
            if (client == null)
                return null;

            return new ClientDTO()
            {
                Id = client.Id,
                Name = client.Name,
                Email = client.Email,
                Role = client.Role
            };
        }
    }
}
